pallets/session-ext/src/manager.rs

```diff
@@ -0,0 +1,123 @@
+use frame_support::log::debug;
+use pallet_session::SessionManager;
+use primitives::EraManager;
+use sp_staking::{EraIndex, SessionIndex};
+use sp_std::{marker::PhantomData, vec::Vec};
+
+use crate::{
+    pallet::{Config, Pallet, SessionValidatorBlockCount},
+    traits::EraInfoProvider,
+};
+
+/// We assume that block `B` ends session nr `S`, and current era index is `E`.
+///
+/// 1. Block `B` initialized
+/// 2. `end_session(S)` is called
+/// *  Based on block count we might mark the session for a given validator as underperformed
+/// *  We update rewards and clear block count for the session `S`.
+/// 3. `start_session(S + 1)` is called.
+/// *  if session `S+1` starts new era we populate totals and unban all validators whose ban expired.
+/// *  if session `S+1` % [`BanConfig::clean_session_counter_delay`] == 0, we
+///    clean up underperformed session counter
+/// 4. `new_session(S + 2)` is called.
+/// *  If session `S+2` starts new era we emit fresh bans events
+/// *  We rotate the validators for session `S + 2` using the information about reserved and non reserved validators.
+
+impl<T> pallet_authorship::EventHandler<T::AccountId, T::BlockNumber> for Pallet<T>
+where
+    T: Config,
+{
+    fn note_author(validator: T::AccountId) {
+        SessionValidatorBlockCount::<T>::mutate(&validator, |count| {
+            *count += 1;
+        });
+    }
+}
+
+/// SessionManager that also fires EraManager functions.
```

Comment:
So this is essentially a wrapper? Could you write this out a bit more explicitly, i.e. that this is _all_ it does? Something like "A wrapper over a SessionManager, that...".

Reply:
Its not all that it does, I have added sentence about it responsibilities